gateway/metrics.go

```diff
@@ -0,0 +1,312 @@
+	}
+}
+
+func (b *ipfsBackendWithMetrics) Get(ctx context.Context, path ImmutablePath) (ContentPathMetadata, *GetResponse, error) {
+	begin := time.Now()
+	name := "IPFSBackend.Get"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, n, err := b.api.Get(ctx, path)
+
+	b.updateApiCallMetric(name, err, begin)
+	return md, n, err
+}
+
+func (b *ipfsBackendWithMetrics) GetRange(ctx context.Context, path ImmutablePath, ranges ...GetRange) (ContentPathMetadata, files.File, error) {
+	begin := time.Now()
+	name := "IPFSBackend.GetRange"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, f, err := b.api.GetRange(ctx, path, ranges...)
+
+	b.updateApiCallMetric(name, err, begin)
+	return md, f, err
+}
+
+func (b *ipfsBackendWithMetrics) GetAll(ctx context.Context, path ImmutablePath) (ContentPathMetadata, files.Node, error) {
+	begin := time.Now()
+	name := "IPFSBackend.GetAll"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, n, err := b.api.GetAll(ctx, path)
+
+	b.updateApiCallMetric(name, err, begin)
+	return md, n, err
+}
+
+func (b *ipfsBackendWithMetrics) GetBlock(ctx context.Context, path ImmutablePath) (ContentPathMetadata, files.File, error) {
+	begin := time.Now()
+	name := "IPFSBackend.GetBlock"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, n, err := b.api.GetBlock(ctx, path)
+
+	b.updateApiCallMetric(name, err, begin)
+	return md, n, err
+}
+
+func (b *ipfsBackendWithMetrics) Head(ctx context.Context, path ImmutablePath) (ContentPathMetadata, files.Node, error) {
+	begin := time.Now()
+	name := "IPFSBackend.Head"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, n, err := b.api.Head(ctx, path)
+
+	b.updateApiCallMetric(name, err, begin)
+	return md, n, err
+}
+
+func (b *ipfsBackendWithMetrics) ResolvePath(ctx context.Context, path ImmutablePath) (ContentPathMetadata, error) {
+	begin := time.Now()
+	name := "IPFSBackend.ResolvePath"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, err := b.api.ResolvePath(ctx, path)
+
+	b.updateApiCallMetric(name, err, begin)
+	return md, err
+}
+
+func (b *ipfsBackendWithMetrics) GetCAR(ctx context.Context, path ImmutablePath) (ContentPathMetadata, io.ReadCloser, <-chan error, error) {
+	begin := time.Now()
+	name := "IPFSBackend.GetCAR"
+	ctx, span := spanTrace(ctx, name, trace.WithAttributes(attribute.String("path", path.String())))
+	defer span.End()
+
+	md, rc, errCh, err := b.api.GetCAR(ctx, path)
+
+	// TODO: handle errCh
+	b.updateApiCallMetric(name, err, begin)
+	return md, rc, errCh, err
```

Comment:
- [ ] TODO

Reply:
:thought_balloon:  @aschmahmann  `updateApiCallMetric` will increase counter for success or failure based on `err`.
Are we missing anything critical if we only check `err` here and ignore `errCh`?